### PR TITLE
Add support for GCOV JSON intermediate format

### LINKED ISSUE #282: Support GCOV intermediate format
For GCC 9.1 I'm planning to come up with JSON format of the intermediate representation.
It's feasible for consumers of the information like lcov and gcovr
Feel free to comment my patch request:
https://gcc.gnu.org/ml/gcc-patches/2018-09/msg01628.html

Example output:
https://users.suse.com/~mliska/tmp/tramp.json

### PR DESCRIPTION
**Pull request by Spacetown**
GCOV introduced a JSON intermediate format with version 9.1.
If the gcov help contains the option `--json-format` this option is used to get compressed JSON files (`.gcov.json.gz`) instead of the human readable files (`.gcov`). The compressed JSON files are only used if the calls are excluded (default) because the call information isn't present in the JSON format yet.

Closes #282

### CODE REVIEW

File: gcovr/formats/gcov/read.py [hunk 1]
```diff
@@ -91,11 +105,13 @@ def read_report(options: Options) -> CovData:
 
 
 def find_existing_gcov_files(search_path, exclude_dirs):
-    """Find .gcov files under the given search path."""
+    """Find .gcov and .json.gz files under the given search path."""
     LOGGER.debug(f"Scanning directory {search_path} for gcov files...")
     gcov_files = list(
         search_file(
-            re.compile(r".*\.gcov$").match, search_path, exclude_dirs=exclude_dirs
+            re.compile(r".*(\.gcov|\.json\.gz)$").match,
```
Comment (Pesa) [inline gcovr/formats/gcov/read.py:112]: The `.json.gz` extension seems rather generic, there's a risk of matching other files unrelated to gcov.

The documentation of `--json-format` says it produces files with extension `.gcov.json.gz` so maybe you can use that instead?

**Comment by fanfeilong** [comment]
.gcov files are useable when debug

**Comment by Spacetown** [comment]
> .gcov files are useable when debug

???

**Comment by Spacetown** [comment]
@Pesa I would be happy if you could review the change so that we can close this PR after a long time.

**Pull request merged**